SignalServiceKit/src/Messages/Interactions/TSInteraction.m

```diff
@@ -16,25 +16,60 @@ + (instancetype)interactionForTimestamp:(uint64_t)timestamp
                         withTransaction:(YapDatabaseReadWriteTransaction *)transaction {
     OWSAssert(timestamp > 0);
 
-    __block int counter = 0;
-    __block TSInteraction *interaction;
+    // Accept any interaction.
+    NSArray<TSInteraction *> *interactions = [self interactionsWithTimestamp:timestamp
+                                                                  withFilter:^(TSInteraction *interaction) {
+                                                                      return YES;
+                                                                  }
+                                                             withTransaction:transaction];
+
+    if (interactions.count < 1) {
+        // TODO: OWSFail()?
```

Comment:
I'm tempted to make this method's return value `nullable` and make the (single) caller robust to failed lookup.

Reply:
Done.